src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -172,30 +195,41 @@ public partial class DistributedApplicationExecutionContext
     {
         public DistributedApplicationExecutionContext(DistributedApplicationExecutionContextOptions options) { }
 
+        public DistributedApplicationExecutionContext(DistributedApplicationOperation operation, string publisherName) { }
+
         public DistributedApplicationExecutionContext(DistributedApplicationOperation operation) { }
 
+        public bool IsInspectMode { get { throw null; } }
+
         public bool IsPublishMode { get { throw null; } }
 
         public bool IsRunMode { get { throw null; } }
 
         public DistributedApplicationOperation Operation { get { throw null; } }
 
+        public string? PublisherName { get { throw null; } set { } }
+
         public System.IServiceProvider ServiceProvider { get { throw null; } }
     }
 
     public partial class DistributedApplicationExecutionContextOptions
     {
+        public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation, string? publisherName = null) { }
+
         public DistributedApplicationExecutionContextOptions(DistributedApplicationOperation operation) { }
 
         public DistributedApplicationOperation Operation { get { throw null; } }
 
+        public string? PublisherName { get { throw null; } }
```

Comment:
Why `string? publisherName = null`? We already have an overload that doesn't take the parameter. So we could drop the `= null`.

Reply:
https://github.com/dotnet/aspire/pull/8533

Reply:
Good catch.

Added to this PR:

https://github.com/dotnet/aspire/pull/8534